server/plugin.go

```diff
@@ -229,7 +229,7 @@ func (p *Plugin) sendSubscribeConfirmationMessage(message string, subscriptionUR
 				"action":           "confirm",
 				"subscription_url": subscriptionURL,
 			},
-			URL: fmt.Sprintf("http://localhost%v/plugins/%v/confirm?token=%v", siteURLPort, manifest.Id, p.configuration.Token),
+			URL: fmt.Sprintf("%v/plugins/%v/confirm?token=%v", siteURLPort, manifest.Id, p.configuration.Token),
```

Comment:
what is the problem with the current approach? it is working fine in the community server

Reply:
The problem is that the `ListenAddress` parameter is could contain an IP address and look like this: `127.0.0.1:8065`. In this case, your generated URL would end up like this: `http://localhost127.0.0.1:8065/plugins...`, which is an invalid address and breaks the confirmation button. Moreover, restricting the confirmation URL to unsecured `http` seems also not the right way.

Reply:
that is not correct, the ListenAddress should contain only the port

Reply:
the plugin runs a subprocess of the mattermost, it is in the same "server" dont need to go out in the network

Reply:
Fine, but then your solution still does not work, because of the mentioned URL generation. My solution works and leaves it to the server/network to determine how to route the confirmation request. Typically it will not go out in the network

Reply:
i'm fine with this change, just asking because there are two ways to do that and both are valid.

can you drop the other commits regarding the go modules? and this PR just for the url change? then if you want you can make another PR to sync the go modules

Reply:
> that is not correct, the ListenAddress should contain only the port
> https://github.com/mattermost/mattermost-server/blob/master/config/default.json#L6

<img width="937" alt="Screenshot 2019-05-28 at 12 04 02" src="https://user-images.githubusercontent.com/172040/58469860-b1a85580-8140-11e9-805a-0d19c79679dc.png">

Reply:
yeah! forgot that, since only use the port